### PR TITLE
A number of boa improvements

### PR DESCRIPTION
**Pull request by rcoreilly**
The examples/boa model has many fixes for longstanding issues that can be ported to the obelisk version.

There are 2 remaining issues before ready to pull:
* GPU is not quite right vs. CPU.
* still needs manual decay of Super and CT to get rid of small number of residual action errors -- do further diagnostics to figure which ones are key.

### CODE REVIEW

File: axon/layer_compute.go [hunk 1]
```diff
@@ -419,6 +440,13 @@ func (ly *Layer) PlusPhasePost(ctx *Context) {
 	switch ly.LayerType() {
 	case MatrixLayer:
 		ly.MatrixGated()
+	case VThalLayer:
+		fallthrough
+	case PTMaintLayer:
+		if ctx.NeuroMod.HasRew.IsTrue() { // if got reward outcome, we clear
+			ly.DecayState(ctx, 1, 1)
+		}
+		// todo: GPU needs to grab the GBuf at start of next trial!
```
Comment (etuleu) [inline axon/layer_compute.go:449]: Is this still todo? Or just a note?

File: examples/boa/approach_env.go [hunk 2]
```diff
@@ -57,7 +59,8 @@ func (ev *Approach) Desc() string {
 // Defaults sets default params
 func (ev *Approach) Defaults() {
 	ev.TimeCost = 0.05
-	ev.Acts = []string{"Forward", "Left", "Right", "Consume"}
+	ev.Acts = []string{"Forward", "Left", "Right", "Consume", "None"}
```
Comment (etuleu) [inline examples/boa/approach_env.go:62]: I like the None action solution

File: examples/boa/boa.go [hunk 3]
```diff
@@ -213,8 +211,11 @@ func (ss *Sim) ConfigNet(net *axon.Network) {
 	// todo: need m1d, driven by smad -- output pathway
 
 	m1 := net.AddLayer2D("M1", nuCtxY, nuCtxX, emer.Hidden)
-	vl := net.AddLayer2D("VL", ny, nAct, emer.Target)  // Action
 	act := net.AddLayer2D("Act", ny, nAct, emer.Input) // Action
+	// vl := net.AddLayer2D("VL", ny, nAct, emer.Target)  // Action
```
Comment (etuleu) [inline examples/boa/boa.go:215]: Why is vl not a target layer anymore?

File: examples/boa/boa.go [hunk 4]
```diff
@@ -894,8 +926,8 @@ func (ss *Sim) ConfigLogs() {
 	axon.LogAddPCAItems(&ss.Logs, ss.Net.AsAxon(), etime.Run, etime.Epoch, etime.Trial)
 
 	axon.LogAddLayerGeActAvgItems(&ss.Logs, ss.Net.AsAxon(), etime.Test, etime.Cycle)
-	ss.Logs.AddLayerTensorItems(ss.Net, "Act", etime.Test, etime.Trial, "TargetLayer")
-	ss.Logs.AddLayerTensorItems(ss.Net, "Act", etime.AllModes, etime.Cycle, "TargetLayer")
+	// ss.Logs.AddLayerTensorItems(ss.Net, "Act", etime.Test, etime.Trial, "TargetLayer")
```
Comment (etuleu) [inline examples/boa/boa.go:929]: Do we not want this logged?

**Review by etuleu — APPROVED** [review]
Left a few comments but nothing major. Lgtm!

File: axon/layer_compute.go (hunk 1 above)
Comment (rcoreilly) [reply axon/layer_compute.go:449]: changed to appropriate note.

File: examples/boa/approach_env.go (hunk 2 above)
Comment (rcoreilly) [reply examples/boa/approach_env.go:62]: ps. it is not rendering "None" as nothing (zeros) on obelisk, but works here..

File: examples/boa/boa.go (hunk 3 above)
Comment (rcoreilly) [reply examples/boa/boa.go:215]: I will delete that old version now -- VL is now a "Pulvinar" layer that predicts the spinal cord "Act" signal, which is either a cleaned-up version of the action it sent down itself (when cortex is driving), or the result of an instinctive action coming from the brainstem.

File: examples/boa/boa.go (hunk 4 above)
Comment (rcoreilly) [reply examples/boa/boa.go:929]: There are no Target layers anymore -- this crashes!  VL was the only target and now it is a Pulvinar.

Comment (rcoreilly) [reply examples/boa/boa.go:929]: and we weren't doing anything with it anyway.

File: examples/boa/approach_env.go (hunk 2 above)
Comment (etuleu) [reply examples/boa/approach_env.go:62]: Oh what is it rendering as? I can take a look at it.

**Pull request merged**